lxc/network_allocations.go

```diff
@@ -15,17 +16,34 @@ type cmdNetworkListAllocations struct {
 	global  *cmdGlobal
 	network *cmdNetwork
 
+	flagFormat      string
 	flagProject     string
 	flagAllProjects bool
 }
 
-func (c *cmdNetworkListAllocations) pretty(input any) string {
-	jsonData, err := json.MarshalIndent(input, "", "\t")
-	if err != nil {
-		return fmt.Sprintf("%v", input)
+func (c *cmdNetworkListAllocations) pretty(allocs []api.NetworkAllocations) error {
+	header := []string{
+		i18n.G("ADDRESSES"),
+		i18n.G("TYPE"),
+		i18n.G("USED BY"),
+		i18n.G("NAT"),
+		i18n.G("HARDWARE ADDRESS"),
+	}
+
+	data := [][]string{}
+	for _, alloc := range allocs {
+		row := []string{
+			strings.Join(alloc.Addresses, " "),
```

Comment:
Should we put each address on a new line within the row?

Reply:
It would be better yes. Has it been done somewhere already ?

Reply:
im not sure, but it seems likely.

Reply:
There is an option we could add in the `cli.RenderTable` function : we could add `table.SetAutoMergeCells(true)` which will does what we want but this will apply for all the calls to this function.

Reply:
We get sth like this with this option activated:

```bash
+----------------------+-------------------------------------------+----------+-------+-------------------+
|       USED BY        |                 ADDRESSES                 |   TYPE   |  NAT  | HARDWARE ADDRESS  |
+----------------------+-------------------------------------------+----------+-------+-------------------+
| /1.0/networks/lxdbr0 | 10.6.105.1/24                             | network  | false |                   |
+                      +-------------------------------------------+          +       +-------------------+
|                      | fd42:3cce:990:a1fd::1/64                  |          |       |                   |
+----------------------+-------------------------------------------+----------+       +-------------------+
| /1.0/instances/u1    | fd42:3cce:990:a1fd:216:3eff:fe04:f095/128 | instance |       | 00:16:3e:04:f0:95 |
+                      +-------------------------------------------+          +       +                   +
|                      | 10.6.105.160/32                           |          |       |                   |
+----------------------+-------------------------------------------+----------+-------+-------------------+

```

Reply:
That's good but you don't mean it will affect all CLI command outputs do you?

Reply:
Only the `instance` entity has an hardware address

Reply:
> That's good but you don't mean it will affect all CLI command outputs do you?

That's what I meant yes. This will apply wherever cli.RenderTable is called. But the cell merge happens only when multiple adjacent rows have the same field in the same column.

Reply:
Which I think is a nice feature anyway as it avoids useless repetition when displaying data.

Reply:
So, maybe we should add this commit in its own separate PR as it affects other commands. Unless we don't want it of course.

Reply:
Yes please

Reply:
@tomponline then you can review this https://github.com/canonical/lxd/pull/12057 first